controllers/managedocs_controller.go

```diff
@@ -248,32 +241,51 @@ func (r *ManagedOCSReconciler) reconcilePhases() error {
 	return nil
 }
 
-func (r *ManagedOCSReconciler) getComponentStatus() v1.ComponentStatusMap {
-	var components v1.ComponentStatusMap
-
-	// Checking the StorageCluster component.
-	var sc ocsv1.StorageCluster
-
-	scNamespacedName := types.NamespacedName{
-		Name:      storageClusterName,
-		Namespace: r.namespace,
+func (r *ManagedOCSReconciler) updateComponentStatus() {
+	// Getting the status of the StorageCluster component.
+	scStatus := &r.managedOCS.Status.Components.StorageCluster
+	if err := r.get(r.storageCluster); err != nil {
+		if r.storageCluster.Status.Phase == "Ready" {
+			scStatus.State = v1.ComponentReady
+		} else {
+			scStatus.State = v1.ComponentPending
+		}
+	} else if errors.IsNotFound(err) {
+		scStatus.State = v1.ComponentNotFound
+	} else {
+		r.Log.V(-1).Info("error getting StorageCluster, setting compoment status to Unknown")
+		scStatus.State = v1.ComponentUnknown
 	}
 
-	err := r.Get(r.ctx, scNamespacedName, &sc)
-	if err == nil {
-		if sc.Status.Phase == "Ready" {
-			components.StorageCluster.State = v1.ComponentReady
+	// Getting the status of the Prometheus component.
+	promStatus := &r.managedOCS.Status.Components.Prometheus
+	if err := r.get(r.prometheus); err == nil {
+		if r.prometheus.Status == nil || r.prometheus.Status.UnavailableReplicas > 0 {
+			promStatus.State = v1.ComponentPending
 		} else {
-			components.StorageCluster.State = v1.ComponentPending
+			promStatus.State = v1.ComponentReady
 		}
 	} else if errors.IsNotFound(err) {
-		components.StorageCluster.State = v1.ComponentNotFound
+		promStatus.State = v1.ComponentNotFound
 	} else {
-		r.Log.Error(err, "error getting StorageCluster, setting compoment status to Unknown")
-		components.StorageCluster.State = v1.ComponentUnknown
+		r.Log.V(-1).Info("error getting Prometheus, setting compoment status to Unknown")
```

Comment:
Is this a way to do warnings? It would be good to have a helper (i.e. r.warning)

Reply:
Yes, Maybe an abstraction above all logging is needed (still not sure about it), but in any case it should not be part of this PR

Reply:
ok